social/app/views/author.py

```diff
@@ -120,10 +123,49 @@ class AuthorListView(generic.ListView):
     def get_queryset(self):
         return Author.objects.all().order_by('-displayName')
 
+    def get_context_data(self, **kwargs):
+        context = super(AuthorListView, self).get_context_data(**kwargs)
+        context['show_remote_find_link'] = Node.objects.filter(local=False)
+        return context
+
 
 class AuthorDetailView(generic.DetailView):
     model = Author
 
+    def get_object(self, queryset=None):
+        author_id = self.kwargs["pk"]
+
+        fetched_new_author = False
+
+        try:
+            author = super(AuthorDetailView, self).get_object(queryset)
+        except Author.DoesNotExist:
+            # No Author found -- so let's go ask our remote Nodes if they've got it
+            for node in Node.objects.filter(local=False):
+                author = node.create_or_update_remote_author(author_id)
```

Comment:
This call looks like it has the potential to short-circuit the requests to remaining nodes (no try-except for the raised exception in the `create_or_update_remote_author` method). It might be better to assume that the author is not found if there's an error returned in the remote node.